johnzon-mapper/src/main/java/org/apache/johnzon/mapper/Mapper.java

```diff
@@ -146,12 +146,31 @@ public JsonValue toStructure(final Object object) {
         if (BigInteger.class.isInstance(object)) {
             return provider.createValue(BigInteger.class.cast(object));
         }
+
         final JsonObjectGenerator objectGenerator = new JsonObjectGenerator(builderFactory);
-        writeObject(object, objectGenerator, null,
-                isDeduplicateObjects(object.getClass()) ? new JsonPointerTracker(null, "/") : null);
+        if (object.getClass().isArray()) {
+            writeObject(objectToList(object), objectGenerator, null,
+                    isDeduplicateObjects(object.getClass()) ? new JsonPointerTracker(null, "/") : null);
+        } else {
+            writeObject(object, objectGenerator, null,
+                    isDeduplicateObjects(object.getClass()) ? new JsonPointerTracker(null, "/") : null);
+        }
         return objectGenerator.getResult();
     }
 
+    private List objectToList(Object object) {
```

Comment:
We already have that kind of logic, therefore question is: cant we reuse it and avoid one array and list allocation OR at least reuse the existing code?